### PR TITLE
Dzelge 66 file up download

### PR DESCRIPTION
**Pull request by dzelge**
When accepting this PR cate-webui will:
- have a file upload/download/browse menu
- A FilesAPI to send file upload and download requests

### CODE REVIEW

File: src/renderer/actions.ts [hunk 1]
```diff
@@ -2395,6 +2378,99 @@ export const OPEN_MESSAGE_BOX = 'OPEN_MESSAGE_BOX';
...
+                console.error(error);
+            });
+    }
+}
+
+/**
+ *
+ * @param processId Process ID
+ */
+
```
Comment (forman) [inline src/renderer/actions.ts:2421]: Remove action doc, or complete. If latter, remove empty lines between docs and action.

File: src/renderer/actions.ts [hunk 2]
```diff
@@ -2395,6 +2378,99 @@ export const OPEN_MESSAGE_BOX = 'OPEN_MESSAGE_BOX';
...
+    return showDialog('fileDownloadDialog');
+}
+
+
+/**
+ *
+ * @param dir Directory to be used
+ * @param file File to upload
+ */
+
```
Comment (forman) [inline src/renderer/actions.ts:2400]: Remove action doc, or complete. If latter, remove empty line between docs and action.

File: src/renderer/actions.ts [hunk 3]
```diff
@@ -2395,6 +2378,99 @@ export const OPEN_MESSAGE_BOX = 'OPEN_MESSAGE_BOX';
...
+           .then(() => {
+               showToast({type: 'success', text: 'Zip ready for download.'});
+           })
+           .catch((error) => {
+               showToast({type: 'error', text: error.toString()});
+               console.error(error);
+           });
+
+
+        //     api.registerProcess(webAPIServiceURL)
```
Comment (forman) [inline src/renderer/actions.ts:2454]: Remove out-commented code entirely or tell readers here, why it should be kept.

File: src/renderer/containers/FileDownloadDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import { DialogState, State } from '../state';
+import { ModalDialog } from '../components/ModalDialog';
+import { connect, DispatchProp } from 'react-redux';
+import * as actions from '../actions';
+import * as selectors from '../selectors';
+import { AnchorButton, ControlGroup, Intent } from "@blueprintjs/core";
```
Comment (forman) [inline src/renderer/containers/FileDownloadDialog.tsx:7]: Move to top. See above.

File: src/renderer/containers/FileDownloadDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import { DialogState, State } from '../state';
```
Comment (forman) [inline src/renderer/containers/FileDownloadDialog.tsx:2]: Move  imports of external modules to top. Empty line please between imports of external modules and own imports.

Comment (forman) [inline src/renderer/containers/FileDownloadDialog.tsx:2]:
```suggestion
import * as React from 'react';
import { connect, DispatchProp } from 'react-redux';
import { AnchorButton, ControlGroup, Intent } from "@blueprintjs/core";

import { DialogState, State } from '../state';
```

File: src/renderer/containers/FileDownloadDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+import { ModalDialog } from '../components/ModalDialog';
+import { connect, DispatchProp } from 'react-redux';
```
Comment (forman) [inline src/renderer/containers/FileDownloadDialog.tsx:4]: Move to top. See above.

File: src/renderer/containers/FileDownloadDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+import * as actions from '../actions';
+import * as selectors from '../selectors';
+import { AnchorButton, ControlGroup, Intent } from "@blueprintjs/core";
+import { showFileOpenDialog } from "../actions";
+import { OpenDialogResult } from "../components/desktop/types";
+import { TextField } from "../components/field/TextField";
+
```
Comment (forman) [inline src/renderer/containers/FileDownloadDialog.tsx:11]: Two lines between imports and actual code.

I wonder if we can configure `.editorconfig` to automate this.

File: src/renderer/containers/FileUploadDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,160 @@
+import * as React from 'react';
+import { DialogState, State } from '../state';
+import { ModalDialog } from '../components/ModalDialog';
+import { connect, DispatchProp } from 'react-redux';
```
Comment (forman) [inline src/renderer/containers/FileUploadDialog.tsx:4]: Move to top. See above.

File: src/renderer/containers/FileUploadDialog.tsx [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+import * as actions from '../actions';
+import * as selectors from '../selectors';
+import Dropzone from 'react-dropzone';
+import { CSSProperties } from "react";
```
Comment (forman) [inline src/renderer/containers/FileUploadDialog.tsx:8]: Move to top. See above.

File: src/renderer/containers/FileUploadDialog.tsx [hunk 10]
```diff
@@ -0,0 +1,160 @@
+import * as React from 'react';
+import { DialogState, State } from '../state';
+import { ModalDialog } from '../components/ModalDialog';
+import { connect, DispatchProp } from 'react-redux';
+import * as actions from '../actions';
+import * as selectors from '../selectors';
+import Dropzone from 'react-dropzone';
```
Comment (forman) [inline src/renderer/containers/FileUploadDialog.tsx:7]: Move to top. See above.

File: src/renderer/containers/FileUploadDialog.tsx [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+import { CSSProperties } from "react";
+import { AnchorButton, ControlGroup, Intent } from "@blueprintjs/core";
```
Comment (forman) [inline src/renderer/containers/FileUploadDialog.tsx:9]: Move to top. See above.

File: src/renderer/containers/FileUploadDialog.tsx [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+import { showDirectorySelectDialog } from "../actions";
+import { TextField } from "../components/field/TextField";
+
```
Comment (forman) [inline src/renderer/containers/FileUploadDialog.tsx:12]: Two empty lines.

File: src/renderer/containers/FilesMenu.tsx [hunk 13]
```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { connect, Dispatch } from 'react-redux';
+import { Menu, MenuDivider, MenuItem } from '@blueprintjs/core';
```
Comment (forman) [inline src/renderer/containers/FilesMenu.tsx:3]: Insert empty line.

File: src/renderer/webapi/apis/FilesApi.ts [hunk 14]
```diff
@@ -0,0 +1,72 @@
+import { callBlobApi, callJsonApi } from "./callHttpApi";
+import { WebAPIClient } from "../WebAPIClient";
+import * as path from "path";
```
Comment (forman) [inline src/renderer/webapi/apis/FilesApi.ts:3]: Is this a node module? Then we must not use it in the UI! If this is ok, move it to top.

**Comment by forman** [comment]
Please also add labels and fix spacing:

1. Label: *Target directory:*
2. Label: *Source files:*

image

**Comment by forman** [comment]
image

**Comment by forman** [comment]
@dzelge please here too fix spacing and insert label *Source files or directories:*

image

**Comment by dzelge** [comment]
Surpassed by #79

**Pull request closed without merging**